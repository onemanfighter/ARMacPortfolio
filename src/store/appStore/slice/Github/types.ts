export type GithubState = {
  defaultProfile: string;
  currentProfile: string;
};

export interface GithubAppAction {
  setCurrentProfile: (profile: string) => void;
}

export type GithubStateSlice = GithubState & GithubAppAction;
