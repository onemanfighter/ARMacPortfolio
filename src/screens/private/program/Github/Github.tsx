import { GithubProps } from './type';

const Github = (props: GithubProps) => {
  return (
    <div>
      <h1>Github</h1>
    </div>
  );
};

export default Github;
