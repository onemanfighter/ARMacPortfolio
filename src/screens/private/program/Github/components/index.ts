export { default as ProjectList1 } from './ProjectList1';
export { default as ProjectList2 } from './ProjectList2';
