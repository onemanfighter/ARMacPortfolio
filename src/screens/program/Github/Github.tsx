import { GithubProps } from './type';

const Home = (props: GithubProps) => {
  return (
    <div>
      <h1>Home</h1>
    </div>
  );
};

export default Home;
