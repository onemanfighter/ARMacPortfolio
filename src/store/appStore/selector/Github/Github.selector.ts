import { AppStoreState } from '../../appStore';

const githubSelector = (state: AppStoreState) => ({
  currentProfile: state.Github.currentProfile,
  defaultProfile: state.Github.defaultProfile,
});

export { githubSelector };
