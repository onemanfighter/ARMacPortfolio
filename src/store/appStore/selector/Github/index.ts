export { githubSelector } from './Github.selector';
