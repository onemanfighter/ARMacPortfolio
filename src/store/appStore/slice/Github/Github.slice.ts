import { AppStoreSlice } from '../../appStore';
import { GithubState, GithubStateSlice } from './types';

const defaultGithubState: GithubState = {
  defaultProfile: 'onemanfighter',
  currentProfile: 'onemanfighter',
};

const createGithubSlice: AppStoreSlice<GithubStateSlice> = (set) => ({
  ...defaultGithubState,
  setCurrentProfile: (profile: string) =>
    set(() => ({ currentProfile: profile })),
});

export default createGithubSlice;
