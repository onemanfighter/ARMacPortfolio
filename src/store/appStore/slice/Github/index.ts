export { default as createGithubSlice } from './Github.slice';
export type { GithubStateSlice, GithubState } from './types';
