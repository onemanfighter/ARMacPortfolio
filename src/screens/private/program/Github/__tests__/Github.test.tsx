import { render } from '@testing-library/react';
import Github from '../Github';

describe('Github', () => {
  it('should render correctly to match snapshot', () => {
    const { container } = render(<Github />);

    expect(container).toMatchSnapshot();
  });
});
