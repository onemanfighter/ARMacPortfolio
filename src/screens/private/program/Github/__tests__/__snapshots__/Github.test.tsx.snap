// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github should render correctly to match snapshot 1`] = `
<div>
  <iframe
    aria-label="github"
    height="100%"
    src="https://github1s.com/onemanfighter/ar-mac-portfolio"
    width="100%"
  />
</div>
`;
